### ISSUE TITLE
private_network interface not onlined in initial `vagrant up` (regression in 1.9.1)

### ISSUE THREAD

**Issue by bdhess** [mitchellh/vagrant#8269]
Interface of type `private_network` is not onlined in a CentOS guest with Vagrant 1.9.1.  This is a regression, as this worked successfully with Vagrant 1.9.0.

### Vagrant version
FAILURE in Vagrant 1.9.1
WORKS in Vagrant 1.9.0

### Host operating system
RHEL 7.3 (`Linux 3.10.0-514.6.1.el7.x86_64 #1 SMP Sat Dec 10 11:15:38 EST 2016 x86_64 x86_64 x86_64 GNU/Linux`)

### Guest operating system
CentOS 7.3

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "boxcutter/centos73"
  config.vm.network "private_network", ip: "10.20.0.101"

  route = "10.20.1.0/24 via 10.20.0.100"
  config.vm.provision "shell", inline: <<-SHELL
    echo '#{route}' >> /etc/sysconfig/network-scripts/route-eth1
    ip route add #{route}
  SHELL
end
```

### Debug output
https://gist.github.com/bdhess/b9ed1ca175c49e39d8e77c51e51ac2cb

### Expected behavior
Route should be added and SSH connection should terminate successfully

### Actual behavior
Route didn't get added and SSH connection terminated abruptly.  SSHing into the machine, it appears that the `eth1` interface is not online (can be worked around by manually running `ifup eth1` in the guest).

### Steps to reproduce
1.  Run `vagrant up` with the supplied Vagrantfile and Vagrant 1.9.1
2. ???
3. Profit

### References
N/A

**Comment by jpic** [comment]
Not for me:

```
[vagrant@compute ~]$ ifup eth1
/usr/sbin/ifup: configuration for eth1 not found.
Usage: ifup <configuration>
```

More in gist: https://gist.github.com/jpic/458f3a3850d40e6e7304378091055895

**Comment by jpic** [comment]
Oops, that works though: sudo ifup enp0s8
Thank you so much @bdhess !

**Comment by chrisroberts** [comment]
Hi! This was fixed via #8148 and will be in the next release. Thanks!

**Comment by jpic** [comment]
Confirming fixed with master, thanks !
